nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -142,6 +142,16 @@ class BtServiceNode : public BT::ActionNodeBase
   {
   }
 
+  /**
+   * @brief Function to perform some user-defined operation upon successful
+   * completion of the action. Could put a value on the blackboard.
+   * @return BT::NodeStatus Returns SUCCESS by default, user may override return another value
+   */
+  virtual BT::NodeStatus on_success()
```

Comment:
How do you feel about on_completion instead? It might make more sense from a application developer's perspective since you don't want to imply that the service necessarily did succeed just by returning

Reply:
I totally agree with you. I just named it `on_success()` to be consistent with the implementation in other BT-Nodes, like the BtActionNode: https://github.com/ros-planning/navigation2/blob/2035819e0d69fb2e5aea80dd06f668911dca9048/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L146
Therefore, a change there, might be necessary to keep the implementations consistent ?

Reply:
I changed it to on_completion in the new commit.